src/test/java/picard/sam/FastqToSamTest.java

```diff
@@ -177,6 +181,67 @@ public void testEmptyFastqAllowed() throws IOException {
         convertFile(emptyFastq, null, FastqQualityFormat.Illumina, false, false, true);
     }
 
+    @Test
+    public void testStreamInputWithoutQuality() throws IOException {
+        ByteArrayOutputStream stderrStream = new ByteArrayOutputStream();
+        PrintStream newStderr = new PrintStream(stderrStream);
+        PrintStream oldStderr = System.err;
+        System.setErr(newStderr);
+
+        final File tmpFile = File.createTempFile("empty", ".sam");
+
+        final String[] args = {
+                "FASTQ=/dev/stdin",
+                "SAMPLE_NAME=sample001",
+                "OUTPUT=" + tmpFile
+        };
+
+        Assert.assertEquals(runPicardCommandLine(args), 1);
+        System.setErr(oldStderr);
+        Assert.assertTrue(stderrStream.toString().endsWith("QUALITY_FORMAT must be specified when input is not a regular file\n"));
+    }
+
+    @Test
+    public void testStreamInput() throws IOException {
+        final File output = newTempSamFile("stdin");
+        String[] command = {
+                "/bin/bash",
+                "-c",
+                "echo -ne '" +
+                "@ERR194147.10008417/1\\n" +
```

Comment:
Could you extract this fastq snippet as a variable for clarity?

The newish text block feature is super nice for this sort of thing.
```
        String fastq = """
                       @ERR194147.10008417/1
                       ATTTAATTAAGAAAATGTAAACTAAATGACAGTAGACAGACAAGTATGCCTTTGC
                       +
                       ???????????????????????????????????????????????????????
                       """;
```

Reply:
Nice. Java isn't my first language. I love learning these things!